src/server/command/worldedit/blueprint.zig

```diff
@@ -0,0 +1,201 @@
+		std.log.warn("Failed to open 'blueprints' directory ({s})", .{@errorName(err)});
+		source.sendMessage("#ff0000Failed to open 'blueprints' directory ({s})", .{@errorName(err)});
+		return;
+	};
+	defer blueprintsDir.close();
+
+	blueprintsDir.dir.deleteFile(fileName) catch |err| {
+		std.log.warn("Failed to delete blueprint file '{s}' ({s})", .{fileName, @errorName(err)});
+		source.sendMessage("#ff0000Failed to delete blueprint file '{s}' ({s})", .{fileName, @errorName(err)});
+		return;
+	};
+
+	std.log.info("Deleted blueprint file: {s}", .{fileName});
+	source.sendMessage("#ff0000Deleted blueprint file: {s}", .{fileName});
+}
+
+fn blueprintList(source: *User) void {
+	var blueprintsDir = std.fs.cwd().makeOpenPath("blueprints", .{.iterate = true}) catch |err| {
+		std.log.warn("Failed to open 'blueprints' directory ({s})", .{@errorName(err)});
+		source.sendMessage("#ff0000Failed to open 'blueprints' directory ({s})", .{@errorName(err)});
+		return;
+	};
+	defer blueprintsDir.close();
+
+	var directoryIterator = blueprintsDir.iterate();
+
+	while(directoryIterator.next() catch |err| {
+		std.log.warn("Failed to read blueprint directory ({s})", .{@errorName(err)});
+		source.sendMessage("#ff0000Failed to read blueprint directory ({s})", .{@errorName(err)});
+		return;
+	}) |entry| {
+		if(entry.kind != .file) break;
+		if(!std.ascii.endsWithIgnoreCase(entry.name, ".blp")) break;
+
+		source.sendMessage("#ffffff- {s}", .{entry.name});
+	}
+}
+
+fn blueprintLoad(args: []const []const u8, source: *User) void {
+	if(args.len < 2) {
+		source.sendMessage("#ff0000/blueprint load requires file-name argument.", .{});
+		return;
+	}
+	if(args.len >= 3) {
+		source.sendMessage("#ff0000Too many arguments for /blueprint load. Expected 1 argument, file-name.", .{});
+		return;
+	}
+
+	const fileName: []const u8 = ensureBlueprintExtension(main.stackAllocator, args[1]);
+	defer main.stackAllocator.free(fileName);
+
+	var blueprintsDir = openDir("blueprints") catch |err| {
+		std.log.warn("Failed to open 'blueprints' directory ({s})", .{@errorName(err)});
+		source.sendMessage("#ff0000Failed to open 'blueprints' directory ({s})", .{@errorName(err)});
+		return;
+	};
+	defer blueprintsDir.close();
+
+	const storedBlueprint = blueprintsDir.read(main.stackAllocator, fileName) catch |err| {
+		std.log.warn("Failed to read blueprint file '{s}' ({s})", .{fileName, @errorName(err)});
+		source.sendMessage("#ff0000Failed to read blueprint file '{s}' ({s})", .{fileName, @errorName(err)});
+		return;
+	};
+	defer main.stackAllocator.free(storedBlueprint);
+
+	if(source.worldEditData.clipboard) |oldClipboard| {
+		oldClipboard.deinit(main.globalAllocator);
+	}
+	source.worldEditData.clipboard = Blueprint.load(main.globalAllocator, storedBlueprint) catch |err| {
+		std.log.warn("Failed to load blueprint file '{s}' ({s})", .{fileName, @errorName(err)});
+		source.sendMessage("#ff0000Failed to load blueprint file '{s}' ({s})", .{fileName, @errorName(err)});
```

Comment:
Since you do this everywhere now, how about a helper function to avoid duplicating the messages everywhere?

Reply:
this is not resolved